manifests/params.pp

```diff
@@ -29,9 +29,21 @@
   if $::operatingsystem == 'Ubuntu' {
     if versioncmp($::lsbdistrelease, '8.04') < 1 {
       $init_style = 'debian'
-    } else {
+    } elsif versioncmp($::operatingsystemrelease, '15.04') < 0 {
       $init_style = 'upstart'
     }
+    else {
+      # Introducing fact 'is_systemd' since Ubuntu 15.04 can be systemd OR upstart (defaults to systemd though)
+      # Fact allows us to retrieve the client side value.
+      # We convert to boolean incase some people may have stringify facts turned on or not set in puppet 3.x still.
+      $is_systemd = str2bool($::is_systemd)
```

Comment:
Hey AJ.  
You've been really helpful so far with the responses to my PR.  You mind just peeking at this commit with the working consul upgrade/downgrade (web_ui) not done yet but part way in.  
https://github.com/sfu-rcg/puppet-consul/commit/8fbbf44266768e311cc6cf0ca4874c8aa39d45e4
It tests fine for me.  The Exec { provider => shell } is because staging module doesn't allow us to push path to the exec and I can't tie the test, grep, and rm down to specific directories in the 'onlyif' statement as they are located in different locations in Ubuntu vs CentOS.  provider => shell fixes that up :)